### ISSUE TITLE
MET_INSTALL_DIR vs. MET_BASE 

### ISSUE THREAD

**Issue by georgemccabe** [NCAR/METplus#224]
These seem similar, so why do we have both? MET_INSTALL_DIR is used to get the full path of the MET executables when calling them. MET_BASE is used to set an environment variable of the same name.

**Comment by georgemccabe** [comment]
From email conversation on metplus-dev, it is not recommended to set MET_BASE environment variable unless you have to. MET_BASE can be referenced in MET config files without the env variable being set. We should remove the setting of this environment variable from any ush script and remove all instances from the METplus config files.

**Comment by georgemccabe** [comment]
MET_BASE is now only found in the CyclonePlotter and TCMPRPlotter. @bikegeek , could you take these references out and ensure that it doesn't break anything?

**Comment by bikegeek** [comment]
affected files for cyclone plotter wrapper: 

1)  master_metplus.py (uncomment import statement for cyclone plotter now that we have access to the necessary plotting libraries)

2) cyclone_plotter.conf (replaced MET_BASE with MET_INSTALL_DIR/share/met)
3) Added README.rst

**Comment by georgemccabe** [comment]
I'm confused, I still see the METplus config variable MET_BASE used in multiple wrappers and test cases besides the files listed above. I also see the environment variable being set in multiple wrappers.

(metplus_env) mccabe@eyewall:/d1/mccabe/METplus.new$ search . MET_BASE | grep get
././internal_tests/test_extract_tiles_unittest.py:            os.environ['MET_BASE'] = CONFIG_INST.getdir('MET_BASE')
././internal_tests/test_series_by_init.py:            os.environ['MET_BASE'] = CONFIG_INST.getdir('MET_BASE')
././internal_tests/pytests/plotting/make_plots/test_make_plots_wrapper.py:    assert(c_dict['MET_BASE'] == mp.config.getdir('MET_INSTALL_DIR')
././ush/tcmpr_plotter_wrapper.py:            os.environ['MET_BASE'] = self.config.getdir('MET_BASE')
././ush/tcmpr_plotter_wrapper.py:                os.path.join(self.config.getdir('MET_BASE'), 'Rscripts/plot_tcmpr.R')
././ush/make_plots_wrapper.py:        c_dict['MET_BASE'] = self.config.getstr('dir', 'MET_BASE')
././ush/cyclone_plotter_wrapper.py:            os.environ['MET_BASE'] = self.config.getdir('MET_BASE')

**Comment by georgemccabe** [comment]
It is also set in parm/metplus_config/metplus_system.conf

MET_BASE = {MET_INSTALL_DIR}/share/met

**Comment by bikegeek** [comment]
Sorry, I thought this issue was directed just at the cyclone plotter wrapper.   I will create new issues that correspond to the code that you found.

**Comment by bikegeek** [comment]
Now I'm confused, on the comment above from October 7 there is a comment:

_MET_BASE is now only found in the CyclonePlotter and TCMPRPlotter. @bikegeek , could you take these references out and ensure that it doesn't break anything?_

So I take it the above comment is incorrect?

**Comment by bikegeek** [comment]
deleting test_extract_tiles_unittest.py . This test was written with the unittest framework on the first version of extract_tiles_wrapper, which has undergone numerous, major refactorings due to change in configuration file parser, wrapper base class, etc.

**Comment by georgemccabe** [comment]
MET_BASE is still found in cyclone_plotter_wrapper.py and tcmpr_plotter_wrapper.py. Both wrappers read the METplus config variable and sets MET_BASE via os.environ. It is also in make_plots_wrapper.py but it is not used for anything. It is still in parm/metplus_config/metplus_system.conf because the wrappers that use it would break if it was removed. 

I didn't realize that it was also used in the internal test directory, but for sake of time those can probably be left in as long as they don't break any tests. We just want to remove MET_BASE removed from any METplus config variable and not used in any wrappers. We also don't want to set the environment variable MET_BASE in any wrappers because it is not recommended as discussed previously (see comment from September 6)

**Comment by bikegeek** [comment]
@malloryprow, can you make the updates to the test_make_plot when you get around to updating your tests?  I know you needed to modify the syntax to support the changes to how fixtures are used.

**Comment by malloryprow** [comment]
Sure! It isn't used in the wrapper. After the rework, I think it is just a "leftover" that got left behind.

**Comment by bikegeek** [comment]
Correct!  In the wrappers I touched, I replaced MET_BASE with
MET_INSTALL_DIR or MET_INSTALL_DIR/share/met, whenever appropriate.

Thanks,
Minna
---------------
Minna Win
National Center for Atmospheric Research
Developmental Testbed Center
Phone: 303-497-8423
Fax:   303-497-8401

**Comment by georgemccabe** [comment]
To clarify
- remove any use of the MET_BASE config variable in METplus configs/wrappers. 
- remove any setting of MET_BASE environment variable (os.environ) in wrappers
- Use of MET_BASE/... in a MET config file is OK, because the MET tools figure out the location of MET automatically. We don't want to reference ${MET_BASE} though.

**Comment by bikegeek** [comment]
sub issues closed:
Github Issue #317 (tcmpr_plotter and test_extract_tiles_unittest.py)
Github Issue #318 (test_series_by_init unittest)
Github Issue #319 (cyclone plotter wrapper)

**Comment by georgemccabe** [comment]
@bikegeek  and @malloryprow  can this issue be closed?

**Comment by bikegeek** [comment]
@bikegeek:  All my sub-issues related to this issue have been closed (see earlier comments). 

@malloryprow, has any of your work been affected by this issue?

**Comment by bikegeek** [comment]
It looks like the plotting tests that Mallory created are gone (and the MET__BASE environment variable was originally set in these tests???). I'm going to close this since all the sub issues associated with this are closed.   Reopen if this issue still persists.

**Comment by georgemccabe** [comment]
I did a grep for MET_BASE and see that the environment variable is still set in TCMPR Plotter

(metplus_env) mccabe@eyewall:/d1/mccabe/METplus/ush$ grep MET_BASE *
grep: plotting_scripts: Is a directory
grep: produtil: Is a directory
grep: __pycache__: Is a directory
tcmpr_plotter_wrapper.py:        if 'MET_BASE' not in os.environ:
tcmpr_plotter_wrapper.py:            self.logger.debug("Setting environment variable MET_BASE to {}".format(met_base))
tcmpr_plotter_wrapper.py:            os.environ['MET_BASE'] = met_base

I also see that the environment variable is used in the SystemConfiguration section of instructions:
https://ncar.github.io/METplus/Users_Guide/systemconfiguration.html#config-quick-start-example

csh: setenv RSCRIPTS_BASE $MET_BASE/scripts/Rscripts
bash: export RSCRIPTS_BASE $MET_BASE/scripts/Rscripts

**Comment by georgemccabe** [comment]
It looks like we can also now remove MET_BASE from the parm/metplus_config/metplus_system.conf since all references of that METplus config are now removed.

**Comment by georgemccabe** [comment]
TCMPR Plotter wrapper calls share/met/Rscripts/plot_tcmpr.R which will exit if MET_BASE is not set in the environment. I added the setting of MET_BASE back to the wrapper in a rush to get the wrapper to run correctly before the release but forgot to update this ticket with more info. This use case could not have run successfully with the block to set MET_BASE removed unless it was already set in the user's environment. 

It seems to me that if possible we should avoid setting MET_BASE in the wrapper (remove the code block I added back) and modify the R script to use $MET_BASE if it is set and if not find the location of MET_BASE is based on its current location (i.e. go up 2 directories) and set it automatically.
